src/pyasm/prod/service/api_xmlrpc.py

```diff
@@ -924,6 +924,46 @@ def log(my, ticket, level, message, category="default"):
         return True
 
 
+
+
+    #
+    # Preferences
+    #
+    @xmlrpc_decorator
+    def get_preference(my, ticket, key):
+        '''Get the users preference for this project
+
+        @params
+        ticket - authentication ticket
+        key - unique key to identify preference
+
+        @return
+        current value of preference
+        '''
+        project_code = Project.get_project_code()
```

Comment:
This probably needs to be thought out more.  Basically, the original implementation was not ideal.  This is because 99% of the time you want a preference for the current project.  So, right now, a developer has to know that a particular preference is project specific and if it is, find the current project and pass it in explicitly.

Maybe the underlying function should look at the current project first, then cascade global.  Only when a project is explicitly set, then is should use that project specific setting only (not sure if is should cascade at that point).